lib/response_bank/response_cache_handler.rb

```diff
@@ -145,16 +145,14 @@ def serve_from_cache(cache_key_hash, match_entity_tag = "*", cache_age_tolerance
         # regen
         @headers.merge!(headers)
 
-        unless @headers['Content-Encoding']
-          ResponseBank.log("Cache hit, but missing content-encoding in the cache value headers")
-          return
-        end
-
-        # if a cache key hit and client doesn't match encoding, return the raw body
-        if !@env['HTTP_ACCEPT_ENCODING'].to_s.include?(@headers['Content-Encoding'])
-          ResponseBank.log("uncompressing payload for client as client doesn't require encoding")
-          body = ResponseBank.decompress(body, @headers['Content-Encoding'])
-          @headers.delete('Content-Encoding')
+        if @headers['Content-Encoding']
+          if !@env['HTTP_ACCEPT_ENCODING'].to_s.include?(@headers['Content-Encoding'])
+            ResponseBank.log("uncompressing payload for client as client doesn't require encoding")
+            body = ResponseBank.decompress(body, @headers['Content-Encoding'])
+            @headers.delete('Content-Encoding')
+          end
+        else
+          ResponseBank.log("Cache hit, but missing content-encoding in the cache value headers, maybe because a 304 or 404 response")
```

Comment:
```suggestion
          ResponseBank.log("Cache hit, but missing content-encoding in the cache value headers, maybe because of a 304 or 404 response")
```